### ISSUE TITLE
seniority-calc-appgyver-db-deployer deploy failed

### ISSUE THREAD

**Issue by MattiaRedaelli** [SAP-samples/successfactors-extension-calculate-employee-seniority#12]
hello, could someone tell me what caused this error?
![167637435-dde38f48-329a-432e-9ea1-e2aea99283e2](https://user-images.githubusercontent.com/102582917/168024405-b58a2438-e8da-4fb4-8991-9bb1d829f3bb.png)

i solved the various compatibility problems and i updated npm, but i don't understand why if i look at the service log files it tells me that they are not up to date
[seniority-calc-appgyver-db-deployer-2022-05-10 13_17_47.368+0000.txt](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8661840/seniority-calc-appgyver-db-deployer-2022-05-10.13_17_47.368%2B0000.txt)

if you also need to load the files of the terminal operation
[seniority-calc-appgyver-db-deployer.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8661858/seniority-calc-appgyver-db-deployer.log)
[seniority-calc-appgyver-srv.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8661859/seniority-calc-appgyver-srv.log)
[OPERATION.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8661860/OPERATION.log)

**Comment by MattiaRedaelli** [comment]
edit: I don't know if they are linked as problems (but I think so, so I add it here) but when I open "EmployeeProfile" the value field is empty

![image](https://user-images.githubusercontent.com/102582917/168024268-fcd48785-4c6a-4170-9804-5c60692ec59d.png)

**Comment by jmsrpp** [comment]
Hi @MattiaRedaelli - Root cause is this error:

Error: com.sap.hana.di.tabledata: The "include_filter" definitions match with 4 records that already exist in the "SAP_CAPIRE_SENIORITYCALC_EMPLOYEE" table

It is complaining that the sample data we provided with the project is already in your database tables. You can just delete the csv files from srv/external/data and the problem should be resolved. We also return an empty array/object if there are no matching records between the Employee table (in HANA Cloud and populated during hire/rehire/terminate events) and the same userId on the SFSF side.

**Comment by MattiaRedaelli** [comment]
Hi @jmsrpp - after deleting the csv files and running "mbt build" this error appears: 

[2022-05-17 09:09:32]  INFO Cloud MTA Build Tool version 1.2.7
[2022-05-17 09:09:32]  INFO generating the "Makefile_20220517090932.mta" file...
[2022-05-17 09:09:32]  INFO done
[2022-05-17 09:09:32]  INFO executing the "make -f Makefile_20220517090932.mta p=cf mtar= strict=true mode=" command...
[2022-05-17 09:09:32]  INFO validating the MTA project
[2022-05-17 09:09:32]  INFO running the "before-all" build...
[2022-05-17 09:09:32]  INFO executing the "npm set registry=https://registry.npmjs.org/ --production" command...
npm WARN config production Use `--omit=dev` instead.
[2022-05-17 09:09:33]  INFO executing the "npm install --production" command...
npm WARN config production Use `--omit=dev` instead.
.
up to date, audited 245 packages in 814ms

48 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
[2022-05-17 09:09:34]  INFO executing the "npx -p @sap/cds-dk cds build --production" command...
.[cds] - determining build tasks for project [/home/user/projects/successfactors-extension-calculate-employee-seniority].
[cds] - the following build tasks will be executed
[cds] -    {
     "build": {
       "target": "gen",
       "tasks": [
         {"for":"hana", "src":"db", "options":{"model":["db","srv","srv/external/PhotoUser"]}},
         {"for":"node-cf", "src":"srv", "options":{"model":["db","srv","srv/external/PhotoUser"]}}
       ]
     }
   }

[cds] - building project [/home/user/projects/successfactors-extension-calculate-employee-seniority], clean [true]
[cds] - cds [5.9.5], compiler [2.13.8], home [/home/user/projects/successfactors-extension-calculate-employee-seniority/node_modules/@sap/cds]

[cds] - done >
[cds] - [ERROR] srv/emp-service.cds:21:49-62: Artifact “PhotoUser.EmpEmployment” has not been found (in entity:“EmployeeService.EmpEmployment”)
[cds] - [ERROR] srv/emp-service.cds:22:13-19: Element or variable “userId” has not been found (in entity:“EmployeeService.EmpEmployment”/element:“userId”)
[cds] - [ERROR] srv/emp-service.cds:22:21-35: Element or variable “lastDateWorked” has not been found (in entity:“EmployeeService.EmpEmployment”/element:“lastDateWorked”)
[cds] - [ERROR] srv/emp-service.cds:22:37-50: Element or variable “seniorityDate” has not been found (in entity:“EmployeeService.EmpEmployment”/element:“seniorityDate”)
[cds] - [ERROR] srv/emp-service.cds:22:52-59: Element or variable “userNav” has not been found (in entity:“EmployeeService.EmpEmployment”/element:“userNav”)
[cds] - [ERROR] srv/emp-service.cds:32:15-28: Artifact “PhotoUser.EmpEmployment” has not been found (in entity:“EmployeeService.EmployeeProfile”)
[ERROR] srv/emp-service.cds:21:49-62: Artifact “PhotoUser.EmpEmployment” has not been found (in entity:“EmployeeService.EmpEmployment”)
[ERROR] srv/emp-service.cds:22:13-19: Element or variable “userId” has not been found (in entity:“EmployeeService.EmpEmployment”/element:“userId”)
[ERROR] srv/emp-service.cds:22:21-35: Element or variable “lastDateWorked” has not been found (in entity:“EmployeeService.EmpEmployment”/element:“lastDateWorked”)
[ERROR] srv/emp-service.cds:22:37-50: Element or variable “seniorityDate” has not been found (in entity:“EmployeeService.EmpEmployment”/element:“seniorityDate”)
[ERROR] srv/emp-service.cds:22:52-59: Element or variable “userNav” has not been found (in entity:“EmployeeService.EmpEmployment”/element:“userNav”)
[ERROR] srv/emp-service.cds:32:15-28: Artifact “PhotoUser.EmpEmployment” has not been found (in entity:“EmployeeService.EmployeeProfile”)

[2022-05-17 09:09:36] ERROR the "before-all"" build failed: could not execute the "npx -p @sap/cds-dk cds build --production" command: exit status 1
make: *** [Makefile_20220517090932.mta:28: pre_build] Error 1
Error: could not build the MTA project: could not execute the "make -f Makefile_20220517090932.mta p=cf mtar= strict=true mode=" command: exit status 2

**Comment by jmsrpp** [comment]
@MattiaRedaelli - Any chance you accidentally deleted ./external/PhotoUser.csn? Those errors are stating that the underlying model artifacts/elements are missing.

**Comment by MattiaRedaelli** [comment]
@jmsrpp - I just checked and the file is present, if there was any error I will send it to you
[PhotoUser.txt](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8711374/PhotoUser.txt)

**Comment by jmsrpp** [comment]
@MattiaRedaelli maybe so but it is a totally different file from the one here:

https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/blob/sfsf-mobile-appgyver/srv/external/PhotoUser.csn

Please update and confirm.

**Comment by MattiaRedaelli** [comment]
@jmsrpp I ran the update and now the "mbt build" command works, but the initial error apparently remains the same, I show you the logs

[seniority-calc-appgyver-srv.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8714501/seniority-calc-appgyver-srv.log)
[OPERATION.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8714502/OPERATION.log)
[seniority-calc-appgyver-db-deployer.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8714504/seniority-calc-appgyver-db-deployer.log)

**Comment by jmsrpp** [comment]
@MattiaRedaelli - Root cause is now about the table data again:

Error: com.sap.hana.di.tabledata: The "include_filter" definitions match with 6 records that already exist in the "SAP_CAPIRE_SENIORITYCALC_EMPLOYEE" table

Check out this answer, where the solution was just to truncate the table. You can do this by clicking the 3 boxes icon next to your HDI container in BAS. Then, find the SAP_CAPIRE_SENIORITYCALC_EMPLOYEE table and run a truncate statement on it.

https://answers.sap.com/questions/13168093/sap-cap-cds-deploy-import-csv.html

**Comment by MattiaRedaelli** [comment]
@jmsrpp - I'm sorry, but I didn't understand how I should access the table. I tried to look for it in the files, but without success and I looked for an icon with three boxes, but the only one is that of extensions. I try to follow the question and manually delete the fields

edit: I manually deleted all 6 "sap / *" fields in the "package.json" file, but the error remains. If I understood correctly and made no other mistakes, these are the logs

[seniority-calc-appgyver-srv.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8752475/seniority-calc-appgyver-srv.log)
[OPERATION.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8752473/OPERATION.log)
[seniority-calc-appgyver-db-deployer.log](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8752474/seniority-calc-appgyver-db-deployer.log)

**Comment by jmsrpp** [comment]
<img width="453" alt="image" src="https://user-images.githubusercontent.com/6976027/169848780-20ab2acc-0bab-490f-a103-96d83901983e.png">

You can enter the HDI container using this button and then execute SQL to clear out the offending table.

**Comment by MattiaRedaelli** [comment]
@jmsrpp - ok, I couldn't find that icon because I didn't have it, now I'll try again

edit: error "The HDI Container cannot be opened. Bind the Target container to an HDI Container Service and try again", i'll try to fix it. I tried to upgrade HANA Cloud but it didn't work.

**Comment by MattiaRedaelli** [comment]
@jmsrpp - update: I managed to connect the db and execute the truncate statement, but if I click on "EmployeeProfile" it returns me "502 Bad Gateway: Registered endpoint failed to handle the request.", although (except add the exception as indicated in the point 3 of the mission) I did everything. is this correct and should I add the exception or is it an error?
how do i add the exception?
[seniority-calc-appgyver-srv-2022-05-30 08_22_42.681+0000.txt](https://github.com/SAP-samples/successfactors-extension-calculate-employee-seniority/files/8796783/seniority-calc-appgyver-srv-2022-05-30.08_22_42.681%2B0000.txt)
